packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -316,9 +316,12 @@ export function createRootHooks<
       {
         ...opts,
         mutationKey: mutationKey,
-        mutationFn: (input) => {
-          return client.mutation(...getClientArgs([path, { input }], opts));
-        },
+        mutationFn:
+          opts?.mutationFn ??
+          defaultOpts?.mutationFn ??
```

Comment:
Probably not needed / good to allow overriding the mutationFn for alllllll mutations? 🤔 

If so, it should probably be more in the direction of [our `overrides:`-option](https://trpc.io/docs/client/react/useUtils#invalidate-full-cache-on-every-mutation)?